src/lib/classes/lsp3-universal-profile.spec.ts

```diff
@@ -60,6 +60,41 @@ describe('LSP3UniversalProfile', () => {
       expect(data[0].startsWith('0x6f357c6a')).toBe(true);
     });
   });
+  describe('Deploying with LSP3Profile Metadata with specified IPFS client options', () => {
+    let signer;
+    let universalProfile;
+    let keyManager;
+
+    beforeAll(async () => {
+      signer = signers[0];
+
+      const { ERC725Account, KeyManager } = (await lspFactory.LSP3UniversalProfile.deploy(
+        {
+          controllerAddresses: ['0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266'],
+          lsp3Profile: lsp3ProfileJson,
+        },
+        {
+          uploadOptions: {
+            ipfsClientOptions: { host: 'ipfs.infura.io', port: 5001, protocol: 'https' },
+          },
+        }
+      )) as DeployedContracts;
+
+      universalProfile = UniversalProfile__factory.connect(ERC725Account.address, signer);
```

Comment:
Why not using `.attach` here?

```suggestion
      universalProfile = UniversalProfile__factory.attach(ERC725Account.address);
```

Reply:
attach is not recognised well on this class for some reason. I think its okay given attach and connect are the same in some cases the class is reused to set setting data also